orderbook/src/main.rs

```diff
@@ -207,7 +215,17 @@ async fn main() {
             .await
             .unwrap();
 
-    let pool_aggregator = PoolAggregator::from_providers(&pair_providers, &web3).await;
+    let pool_aggregator = PoolAggregator {
+        pool_fetchers: pair_providers
+            .into_iter()
+            .map(|pair_provider| {
+                Arc::new(PoolFetcher {
+                    pair_provider,
+                    web3: web3.clone(),
+                }) as Arc<dyn PoolFetching>
+            })
+            .collect(),
+    };
```

Comment:
I know we removed the `from_providers` method, but wouldn't it be nice to keep all this enclosed in some method implemented for pool aggregator? Was there some special reason for bringing it out? It appears to be doing nearly the same thing as before but with an Arc.

Reply:
The setup is different in the orderbook main and solver main so I think this way makes sense.

Reply:
As you wish!